lms/djangoapps/certificates/tests/test_support_views.py

```diff
@@ -132,6 +138,27 @@ def test_results(self):
         self.assertEqual(retrieved_cert["grade"], unicode(self.CERT_GRADE))
         self.assertEqual(retrieved_cert["status"], self.CERT_STATUS)
         self.assertEqual(retrieved_cert["type"], self.CERT_MODE)
+        self.assertEqual(retrieved_cert["download_url"], self.CERT_DOWNLOAD_URL)
+
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    def test_download_link(self):
+        self.cert.download_url = ''
+        self.cert.save()
+
+        response = self._search(self.STUDENT_USERNAME)
+        self.assertEqual(response.status_code, 200)
+        results = json.loads(response.content)
+
+        self.assertEqual(len(results), 1)
+        retrieved_cert = results[0]
+
+        self.assertEqual(
+            retrieved_cert["download_url"],
+            self.CERT_WEB_VIEW_URL.format(
```

Comment:
I think instead of hardcoding url in `CERT_WEB_VIEW_URL` you should get it via reverse like this
`reverse('certificates:html_view', kwargs={"user_id": self.student.id, "course_id": self.CERT_COURSE_KEY})`